filament/src/Material.cpp

```diff
@@ -378,6 +378,10 @@ Handle<HwProgram> FMaterial::getSurfaceProgramSlow(uint8_t variantKey) const noe
 
     if (Variant(variantKey).hasSkinningOrMorphing()) {
         pb.setUniformBlock(BindingPoints::PER_RENDERABLE_BONES, PerRenderableUibBone::_name);
+        pb.setUniformBlock(BindingPoints::PER_RENDER_PRIMITIVE_MORPHING, PerRenderPrimitiveMorphingUib::_name);
+
+        addSamplerGroup(pb, BindingPoints::PER_RENDER_PRIMITIVE_MORPHING,
```

Comment:
We have a very limited number of samplers on mobile (16) so we probably can't use one just for morphing. This needs to be investigated to see what other sampler could be shared instead.

Reply:
That's fine. I have checked whether exceeding the limted number of samplers on iOS, Android and macOS and there was no issues if we use one sampler for morphing. Also after adding one sampler, all sampler slots are occupied.

Reply:
That is not fine because user materials (or extensions in gltf) may need samplers. We are already at the limit with some of our users.

Reply:
Hm... then there is no more sample binding space? Actually current morphing implementation is not fine as I thought. I want to hear your ideas... or Do you have a plan to detach Sampler from Texture? If we do it that we can binding more texture than 16.

Reply:
I tried to use SSBO but Mali doesn't support binding SSBO on the vertex shader. Also ARM doesn't have a plan to support this. ARM's answer is [here](https://community.arm.com/support-forums/f/graphics-gaming-and-vr-forum/13645/can-a-mali-g71-use-a-ssbo-in-a-vertex-shader). So it's only way uploading morphing data to a texture.